airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -336,6 +336,37 @@ def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mappi
             backoff_handler = default_backoff_handler(max_tries=max_tries, factor=self.retry_factor)
             return backoff_handler(user_backoff_handler)(request, request_kwargs)
 
+    def parse_response_error_message(self, response: requests.Response) -> Optional[str]:
+        # default logic to grab error from common fields
+
+        def _try_get_error(value):
+            if type(value) is str:
+                return value
+            elif type(value) is list:
+                return ", ".join(_try_get_error(v) for v in value)
+            elif type(value) is dict:
+                new_value = (
+                    value.get("message")
+                    or value.get("messages")
+                    or value.get("error")
+                    or value.get("errors")
+                    or value.get("failures")
+                    or value.get("failure")
+                )
+                return _try_get_error(new_value)
+            return None
+
+        try:
+            body = response.json()
```

Comment:
we shouldn't assume the response is JSON

Reply:
Yeah, at the moment I'm returning `None` if a `JSONDecodeError` is encountered... Breaking it out into a separate method for decoding feels like it may cause confusion (this would only be used for parsing error responses but parsing regular responses is left up to you?). I think in this case you would override the method as needed to fit your use case.